### PR TITLE
feat(FX-4440): Pin Collector Profile-sourced price range as the last "recent" price range

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [FX-4440] 

### Demo
https://user-images.githubusercontent.com/3513494/203063717-5cce14ab-ff6b-4257-89d5-6e8f2756f618.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [x] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Pin Collector Profile-sourced price range as the last "recent" price range - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4440]: https://artsyproduct.atlassian.net/browse/FX-4440?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tests.tsx [hunk 1]
```diff
@@ -258,6 +258,44 @@ describe("PriceRangeOptions", () => {
         expect(queryByText("Apply a recent Price Range")).toBeNull()
       })
 
+      it("should render the collector profile-sourced range", () => {
```
Comment (nickskalkin) [inline src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tests.tsx:261]: **suggestion / non-blocking**: I would add a test covering `userPreferredPriceRange !== USER_PREFERRED_DEFAULT_PRICE_RANGE` case, but up to you!

**Review by nickskalkin — DISMISSED** [review]
LGTM 👍 One small non-blocking comment

File: src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tests.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tests.tsx:261]: good suggestions. i'll add test case for it

Comment (dimatretyak) [reply src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tests.tsx]: Done

File: src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tsx [hunk 2]
```diff
@@ -329,3 +334,21 @@ export const PriceRangeOptionsScreen: React.FC<PriceRangeOptionsScreenProps> = (
...
+
+const usePriceRanges = () => {
+  let recentPriceRanges = useRecentPriceRanges()
+  const userPreferredPriceRange = GlobalStore.useAppState((state) => state.userPrefs.priceRange)
+
+  if (userPreferredPriceRange !== USER_PREFERRED_DEFAULT_PRICE_RANGE) {
+    const slicedRecentPriceRanges = recentPriceRanges.slice(0, MAX_SHOWN_RECENT_PRICE_RANGES - 1)
+
+    // Remove duplicate price range that matches collector profile-sourced price
+    recentPriceRanges = slicedRecentPriceRanges.filter((priceRange) => {
```
Comment (dblandin) [inline src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tsx:346]: **question / non-blocking**: As is, does this mean with other recent price ranges, the collector profile-sourced price range would never appear towards the front of the list — even if most recently selected?

Would the other way around be a better experience? If the collector profile-sourced price range is already in the displayable recent set — do not append to the list?

**Review by dblandin — DISMISSED** [review]
LGTM! - left one question regarding the deduplication logic

File: src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tsx:346]: > does this mean with other recent price ranges, the collector profile-sourced price range would never appear towards the front of the list — even if most recently selected

Yes, that's right (if I understood your message correctly). The collector profile-sourced price range will always be displayed last in the list

Comment (dimatretyak) [reply src/app/Components/ArtworkFilter/Filters/PriceRangeOptions.tsx:346]: > f the collector profile-sourced price range is already in the displayable recent set — do not append to the list?

I will discuss this point with Ryan

**Review by nickskalkin — APPROVED** [review]

**Pull request merged**